web-common/src/test/java/edu/unc/lib/boxc/web/common/services/AccessCopiesServiceTest.java

```diff
@@ -171,6 +194,52 @@ public void testDoesNotHaveViewablePdfPidMultipleFileObjects() {
         assertNull(filePid);
     }
 
+    @Test
+    public void testPrimaryObjectHasDownloadUrl() {
+        hasPermissions(mdObjectAudio, true);
+        String downloadUrl = accessCopiesService.getDownloadPath(mdObjectAudio, principals);
+        assertEquals("content/" + mdObjectAudio.getId(), downloadUrl);
+    }
+
+    @Test
+    public void testChildObjectHasDownloadUrl() {
+        hasPermissions(noOriginalFileObj, true);
+        hasPermissions(mdObject, true);
+
+        List<ContentObjectSolrRecord> resultList = Collections.singletonList(mdObject);
+        when(queryResponse.getBeans(ContentObjectSolrRecord.class)).thenReturn(resultList);
+
+        String downloadUrl = accessCopiesService.getDownloadPath(noOriginalFileObj, principals);
+        assertEquals("indexablecontent/" + mdObject.getId(), downloadUrl);
+    }
+
+    @Test
+    public void testDoesNotHasDownloadUrl() {
+        hasPermissions(noOriginalFileObj, true);
+        hasPermissions(mdObjectXml, true);
+
+        List<ContentObjectSolrRecord> resultList = Collections.singletonList(mdObjectXml);
+        when(queryResponse.getBeans(ContentObjectSolrRecord.class)).thenReturn(resultList);
+
+        String downloadUrl = accessCopiesService.getDownloadPath(noOriginalFileObj, principals);
+
+        assertEquals("", downloadUrl);
+    }
+
```

Comment:
Could you add a test to verify that a work that contains multiple files and no primary object doesn't get a download url? And perhaps a test to verify that a work with multiple files and a primary object does get the right one.

Reply:
Not sure how to add a test for a work with multiple files and a primary object, as it should never query for the files if the primary object is set.

Reply:
Okay, probably fine to leave it off. We could sort of set it up (mostly by having the total result count set to 2), but might not make sense to go through the effort at the moment